src/context/AgentContext.tsx

```diff
@@ -0,0 +1,295 @@
+import { showNotification } from "@mantine/notifications";
+import { Ad4mClient, ExceptionType, Link } from "@perspect3vism/ad4m";
+import { ExceptionInfo } from "@perspect3vism/ad4m/lib/src/runtime/RuntimeResolver";
+import { createContext, useEffect, useState } from "react";
+import { AD4M_ENDPOINT } from "../config";
+import { SOURCE_PROFILE, PREDICATE_FIRSTNAME, PREDICATE_LASTNAME } from "../constants/triples";
+import { buildAd4mClient } from "../util";
+
+type State = {
+  url: string;
+  isInitialized: Boolean;
+  isUnlocked: Boolean;
+  loading: boolean;
+  client: Ad4mClient | null;
+  did: string,
+  candidate: string;
+  connected: boolean;
+  connectedLaoding: boolean;
+}
+
+type ContextProps = {
+  state: State;
+  methods: {
+    setUrl: (str: string) => void,
+    resetUrl: () => void
+    handleTrustAgent: (str: string) => void,
+    unlockAgent: (str: string) => void,
+    lockAgent: (str: string) => void,
+    generateAgent: (firstName: string, lastName: string, password: string) => void,
```

Comment:
And this seems too much global state and state methods, do we really want to make the global state such big? It will reduce the maintainance and readable.

Reply:
This is just for Agent, we would have separate context for language, perspective and so on.